### ISSUE TITLE
Question about training data on retrieval task

### ISSUE THREAD

**Issue by wangqian621** [XinhaoMei/WavCaps#14]
First of all, thanks for your code!
I wonder how you train AC+Clotho+WavCaps together, I have seen that different datasets are fed into the “AudioCaptionDataset” module separately in your code .So I actually train them one by one, it coudn't reach the promising result.
Thanks again!

**Comment by jwang1993** [comment]
In the pre-training phase, I also tried AC+Clotho+WavCaps together to train the pretraining model, but the pre-trained model failed to obtain the promising results in the paper. And I tried "gradient accumulation steps" but also failed to achieve the precision in the paper.

**Comment by wangqian621** [comment]
> 

Have you fed them into one dataset and then randomly selected samples for pre-training? I fear that their different format or scale will affect the performance if they are trained together.

**Comment by wangqian621** [comment]
> In the pre-training phase, I also tried AC+Clotho+WavCaps together to train the pretraining model, but the pre-trained model failed to obtain the promising results in the paper. And I tried "gradient accumulation steps" but also failed to achieve the precision in the paper.

Have you fed them into one dataset and then randomly selected samples for pre-training? I fear that their different format or scale will affect the performance if they are trained together.

**Comment by XinhaoMei** [comment]
> First of all, thanks for your code! I wonder how you train AC+Clotho+WavCaps together, I have seen that different datasets are fed into the “AudioCaptionDataset” module separately in your code .So I actually train them one by one, it coudn't reach the promising result. Thanks again!

Hi, 

You need to use the pretrain code, and specify the `json_files` you wan to use in `pretrain.yaml`.

**Comment by XinhaoMei** [comment]
> In the pre-training phase, I also tried AC+Clotho+WavCaps together to train the pretraining model, but the pre-trained model failed to obtain the promising results in the paper. And I tried "gradient accumulation steps" but also failed to achieve the precision in the paper.

Hi,


Can I know your results and hyperparameter settings?

**Comment by jwang1993** [comment]
> > In the pre-training phase, I also tried AC+Clotho+WavCaps together to train the pretraining model, but the pre-trained model failed to obtain the promising results in the paper. And I tried "gradient accumulation steps" but also failed to achieve the precision in the paper.
> 
> Have you fed them into one dataset and then randomly selected samples for pre-training? I fear that their different format or scale will affect the performance if they are trained together.

Yes, I have fed them into one dataset and the code you published has not been modified in any way. I run the pretrain.py  using distributed mode.

**Comment by XinhaoMei** [comment]
> > > In the pre-training phase, I also tried AC+Clotho+WavCaps together to train the pretraining model, but the pre-trained model failed to obtain the promising results in the paper. And I tried "gradient accumulation steps" but also failed to achieve the precision in the paper.
> > 
> > 
> > Have you fed them into one dataset and then randomly selected samples for pre-training? I fear that their different format or scale will affect the performance if they are trained together.
> 
> Yes, I have fed them into one dataset and the code you published has not been modified in any way.

What's your results?

The hyperparameters used used for pretraining are as follows:

{'audio_args': {'f_max': 14000,
                'f_min': 50,
                'hop_length': 320,
                'max_length': 10,
                'mono': True,
                'n_fft': 1024,
                'n_mels': 64,
                'sr': 32000},
 'audio_encoder_args': {'freeze': False,
                        'model': 'htsat',
                        'pretrained': True,
                        'type': 'transformer'},
 'blacklist': '/mnt/fast/nobackup/scratch4weeks/xm00178/WavCaps/data/json_files/blacklist/blacklist_exclude_test_ac.json',
 'data_args': {'batch_size': 128, 'num_workers': 8},
 'device': 'cuda',
 'dist_args': {'distributed': False, 'world_size': 1},
 'embed_regularization': True,
 'embed_size': 1024,
 'json_files': ['/mnt/fast/nobackup/scratch4weeks/xm00178/WavCaps/data/json_files/BBC_Sound_Effects/bbc_pretrain.json',
                '/mnt/fast/nobackup/scratch4weeks/xm00178/WavCaps/data/json_files/FreeSound/fsd_pretrain.json',
                '/mnt/fast/nobackup/scratch4weeks/xm00178/WavCaps/data/json_files/SoundBible/sb_pretrain.json',
                'data/AudioCaps/json_files/train.json',
                'data/Clotho/json_files/train.json',
                '/mnt/fast/nobackup/scratch4weeks/xm00178/WavCaps/data/json_files/AudioSet_SL/as_pretrain.json'],
 'optim_args': {'betas': [0.9, 0.999],
                'eps': 1e-08,
                'lr': 5e-05,
                'momentum': 0.9,
                'optimizer_name': 'adam',
                'warmup_epochs': 2,
                'warmup_steps': 0},
 'queue_size': 5120,
 'resume': False,
 'seed': 20,
 'temp': 0.07,
 'text_encoder_args': {'freeze': False, 'type': 'bert-base-uncased'},
 'training': {'clip_grad': 2,
              'dropout': 0.2,
              'epochs': 15,
              'spec_augmentation': True}}

**Comment by jwang1993** [comment]
> > > > In the pre-training phase, I also tried AC+Clotho+WavCaps together to train the pretraining model, but the pre-trained model failed to obtain the promising results in the paper. And I tried "gradient accumulation steps" but also failed to achieve the precision in the paper.
> > > 
> > > 
> > > Have you fed them into one dataset and then randomly selected samples for pre-training? I fear that their different format or scale will affect the performance if they are trained together.
> > 
> > 
> > Yes, I have fed them into one dataset and the code you published has not been modified in any way.
> 
> What's your results?
> 
> The hyperparameters used used for pretraining are as follows:
> 
> {'audio_args': {'f_max': 14000, 'f_min': 50, 'hop_length': 320, 'max_length': 10, 'mono': True, 'n_fft': 1024, 'n_mels': 64, 'sr': 32000}, 'audio_encoder_args': {'freeze': False, 'model': 'htsat', 'pretrained': True, 'type': 'transformer'}, 'blacklist': '/mnt/fast/nobackup/scratch4weeks/xm00178/WavCaps/data/json_files/blacklist/blacklist_exclude_test_ac.json', 'data_args': {'batch_size': 128, 'num_workers': 8}, 'device': 'cuda', 'dist_args': {'distributed': False, 'world_size': 1}, 'embed_regularization': True, 'embed_size': 1024, 'json_files': ['/mnt/fast/nobackup/scratch4weeks/xm00178/WavCaps/data/json_files/BBC_Sound_Effects/bbc_pretrain.json', '/mnt/fast/nobackup/scratch4weeks/xm00178/WavCaps/data/json_files/FreeSound/fsd_pretrain.json', '/mnt/fast/nobackup/scratch4weeks/xm00178/WavCaps/data/json_files/SoundBible/sb_pretrain.json', 'data/AudioCaps/json_files/train.json', 'data/Clotho/json_files/train.json', '/mnt/fast/nobackup/scratch4weeks/xm00178/WavCaps/data/json_files/AudioSet_SL/as_pretrain.json'], 'optim_args': {'betas': [0.9, 0.999], 'eps': 1e-08, 'lr': 5e-05, 'momentum': 0.9, 'optimizer_name': 'adam', 'warmup_epochs': 2, 'warmup_steps': 0}, 'queue_size': 5120, 'resume': False, 'seed': 20, 'temp': 0.07, 'text_encoder_args': {'freeze': False, 'type': 'bert-base-uncased'}, 'training': {'clip_grad': 2, 'dropout': 0.2, 'epochs': 15, 'spec_augmentation': True}}


My hyperparameters used for pretraining are as follows:
{'audio_args': {'f_max': 14000,
                'f_min': 50,
                'hop_length': 320,
                'max_length': 30,
                'mono': True,
                'n_fft': 1024,
                'n_mels': 64,
                'sr': 32000},
 'audio_encoder_args': {'freeze': False,
                        'model': 'Cnn14',
                        'pretrained': True,
                        'type': 'cnn'},
 'blacklist': 'data/json_files/blacklist/blacklist_exclude_ub8k_esc50_vggsound.json',
 'data_args': {'batch_size': 32, 'num_workers': 8},
 'device': 'cuda',
 'dist_args': {'dist_backend': 'nccl',
               'dist_url': 'tcp://11.33.242.125:15596',
               'distributed': True,
               'gpu': 2,
               'rank': 2,
               'world_size': 6},
 'embed_regularization': True,
 'embed_size': 1024,
 'json_files': ['data/json_files/BBC_Sound_Effects/bbc_final_new.json',
                'data/json_files/FreeSound/fsd_final_new.json',
                'data/json_files/SoundBible/sb_final_new.json',
                'data/json_files/AudioSet_SL/as_final_new.json',
                'data/AudioCaps/json_files/train.json',
                'data/Clotho/json_files/train.json'],
 'optim_args': {'betas': [0.9, 0.999],
                'eps': 1e-08,
                'lr': 5e-05,
                'momentum': 0.9,
                'optimizer_name': 'adam',
                'warmup_epochs': 2,
                'warmup_steps': 0},
 'queue_size': 5120,
 'resume': False,
 'seed': 20,
 'temp': 0.07,
 'text_encoder_args': {'freeze': False, 'type': 'bert-base-uncased'},
 'training': {'clip_grad': 2,
              'dropout': 0.2,
              'epochs': 45,
              'spec_augmentation': True}}

My results is as follows:
![image](https://github.com/XinhaoMei/WavCaps/assets/25115861/afd8d3f8-811a-4f10-92a9-d54afd8f3f0a)

**Comment by wangqian621** [comment]
> > First of all, thanks for your code! I wonder how you train AC+Clotho+WavCaps together, I have seen that different datasets are fed into the “AudioCaptionDataset” module separately in your code .So I actually train them one by one, it coudn't reach the promising result. Thanks again!
> 
> Hi,
> 
> You need to use the pretrain code, and specify the `json_files` you wan to use in `pretrain.yaml`.
Oh，I have ignored it，thanks！！
By the way，I failed to use "load_dataset"func in huggingface to load **Wavcaps** dataset, how can I get it?

**Comment by XinhaoMei** [comment]
> > > First of all, thanks for your code! I wonder how you train AC+Clotho+WavCaps together, I have seen that different datasets are fed into the “AudioCaptionDataset” module separately in your code .So I actually train them one by one, it coudn't reach the promising result. Thanks again!
> > 
> > 
> > Hi,
> > You need to use the pretrain code, and specify the `json_files` you wan to use in `pretrain.yaml`.
> > Oh，I have ignored it，thanks！！
> > By the way，I failed to use "load_dataset"func in huggingface to load **Wavcaps** dataset, how can I get it?

Sorry, it's not supported yet.
